### PR TITLE
Lnl create new platform

### PR DESCRIPTION
**Pull request by jxstelter**
Add board definition for Lunarlake.
This is first PR from the series to enable new ACE 2.0 platform.

Signed-off-by: Jaroslaw Stelter <Jaroslaw.Stelter@intel.com>

### CODE REVIEW

File: soc/xtensa/intel_adsp/ace/include/intel_ace20_lnl/adsp_shim.h [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+
+
+#define ADSP_SHIM_DSPWCTCS_TTIE(c) BIT(8 + (c))
+
+#endif /* _ASMLANGUAGE */
+
+#define ACE_CLKCTL_WOVCRO    BIT(4)	  /* Request WOVCRO clock */
+
+#define SHIM_LDOCTL_HPSRAM_LDO_ON     (3 << 0)
```
Comment (ceolin) [inline soc/xtensa/intel_adsp/ace/include/intel_ace20_lnl/adsp_shim.h:97]: `(3)` ?

File: soc/xtensa/intel_adsp/ace/include/intel_ace20_lnl/adsp_shim.h [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+#define SHIM_LDOCTL_HPSRAM_LDO_BYPASS BIT(0)
+
+#define SHIM_LDOCTL_LPSRAM_LDO_ON     (3 << 2)
+#define SHIM_LDOCTL_LPSRAM_LDO_BYPASS BIT(2)
+
+#define ADSP_DMWBA_ENABLE   BIT(0)
+#define ADSP_DMWBA_READONLY BIT(1)
+
+#define ADSP_CLKCTL_OSC_SOURCE_MASK  (3<<2)
+#define ADSP_CLKCTL_OSC_REQUEST_MASK (~BIT_MASK(28))
```
Comment (ceolin) [inline soc/xtensa/intel_adsp/ace/include/intel_ace20_lnl/adsp_shim.h:107]: `~BIT_MASK` is exactly what you want for mask ?

**Comment by ceolin** [comment]
Nothing wrong popped to me. Looks nice !

**Review by nashif — DISMISSED** [review]

File: soc/xtensa/intel_adsp/ace/include/intel_ace20_lnl/adsp_shim.h (hunk 1 above)
Comment (jxstelter) [reply soc/xtensa/intel_adsp/ace/include/intel_ace20_lnl/adsp_shim.h:97]: This was inherited from MTL config. Although it looks like all definitions SHIM_LDOCTL_* are copied from CAVS configurations and not used even for MTL..

File: soc/xtensa/intel_adsp/ace/include/intel_ace20_lnl/adsp_shim.h (hunk 2 above)
Comment (jxstelter) [reply soc/xtensa/intel_adsp/ace/include/intel_ace20_lnl/adsp_shim.h:107]: Yes, CLKCTL is identical for MTL and LNL.

**Review by nashif — APPROVED** [review]

**Review by tmleman — APPROVED** [review]

**Pull request merged**